### PR TITLE
Features/registration form

### PR DESCRIPTION
**Pull request by nattikim**
Review, please. Had to start it "from scratch". Previous branch has to be deleted.

### CODE REVIEW

File: backend/database/models.go [hunk 1]
```diff
@@ -13,10 +13,14 @@ type Post struct {
 }
 
 type User struct {
-	Id       int
-	Name     string
-	Email    string
-	Password string
+	Id        int
+	Name      string
+	Email     string
```
Comment (maximihajlov) [inline backend/database/models.go:18]: Looks like you did not test it on the existing database. Error 500:

image

Optional fields are NULLABLE, so they can not be saved to standard golang types. Use database types such as [`sql.NullString`](https://pkg.go.dev/database/sql#NullString) instead.

File: backend/server/handlers_auth.go [hunk 2]
```diff
@@ -18,9 +18,13 @@ func (srv *Server) signupHandler(w http.ResponseWriter, r *http.Request) {
 	}
 
 	requestBody := struct {
-		Name     string `json:"name"`
-		Email    string `json:"email"`
-		Password string `json:"password"`
+		Name      string `json:"name"`
+		Email     string `json:"email"`
+		Password  string `json:"password"`
+		FirstName string `json:"first_name"`
```
Comment (maximihajlov) [inline backend/server/handlers_auth.go:24]: We should not always send optional fields. 
```json
{
    "id": 1,
    "name": "user",
    "email": "user@example.com",
    "first_name": "John",
    "last_name": "",
    "age": "",
    "gender": ""
}
```
Should be: 
```json
{
    "id": 1,
    "name": "user",
    "email": "user@example.com",
    "first_name": "John",
}
```

File: backend/database/models.go [hunk 3]
```diff
@@ -13,10 +13,14 @@ type Post struct {
...
+	Password  string
+	FirstName string
+	LastName  string
+	Age       string
```
Comment (maximihajlov) [inline backend/database/models.go:22]: Maybe age should be a number? And maybe it makes more sense to ask for the date of birth instead to keep the age up to date.

File: frontend/src/pages/me.tsx [hunk 4]
```diff
@@ -86,7 +86,18 @@ const UserInfo = () => {
...
-        {"Your email is "} <span className={"text-2xl"}> {user?.email} </span>
+        {"Full name: "}
+        <span className={"text-2xl"}> {user?.first_name + " "} </span>
+        <span className={"text-2xl"}> {user?.last_name} </span>
+      </p>
+      <p>
+        {"Age: "} <span className={"text-2xl"}> {user?.age} </span>
+      </p>
+      <p>
+        {"Gender: "} <span className={"text-2xl"}> {user?.gender} </span>
```
Comment (maximihajlov) [inline frontend/src/pages/me.tsx:97]: Also not sure about gender as a string. Maybe it's better to ask for preferred `pronouns` instead.

File: frontend/src/pages/signup.tsx [hunk 5]
```diff
@@ -86,6 +90,77 @@ const SignupPage: NextPage = () => {
...
+            type={"text"}
+            id={"gender"}
+            className={
+              "bg-gray-50 border border-gray-300 text-gray-900 text-sm rounded-lg focus:ring-blue-500 focus:border-blue-500 block w-full p-2.5 dark:bg-gray-700 dark:border-gray-600 dark:placeholder-gray-400 dark:text-white dark:focus:ring-blue-500 dark:focus:border-blue-500"
+            }
+            onInput={(e) => setGender(e.currentTarget.value)}
+            placeholder={"Gender"}
+          />
+        </div>
+
```
Comment (maximihajlov) [inline frontend/src/pages/signup.tsx:163]: All input fields are mostly the same. It's better to start creating custom tailwind classes now. 

Another option is not to store fields in an array of objects and generate it with for loop. Like [**here**](https://github.com/merpw/forum/blob/main/frontend/src/pages/me.tsx#LL60B1-L73C12):
```tsx
  const tabs = [
    { title: "Your posts", component: <UserPosts /> },
    { title: "Your liked posts", component: <UserLikedPosts /> },
  ]

 ...

<ul className={"flex flex-wrap gap-2 text-2xl mb-3"}>
        {tabs.map(({ title }, key) => (
          <li
            key={key}
            className={
              "cursor-pointer hover:opacity-60 p-1 " +
              (activeTab == key ? "border-b-2 border-b-blue-500" : "")
            }
            onClick={() => setActiveTab(key)}
          >
            {title}
          </li>
        ))}
</ul>
```

**Review by maximihajlov — CHANGES_REQUESTED** [review]

File: backend/server/handlers_auth.go (hunk 2 above)
Comment (sagarishere) [reply backend/server/handlers_auth.go:24]: all fields are required fields. right? 
image

**Review by maximihajlov — CHANGES_REQUESTED** [review]

File: backend/server/test/get_test.go [hunk 6]
```diff
@@ -26,7 +26,12 @@ func TestGet(t *testing.T) {
 
 	cli := testServer.Client()
```
[the remark this answers is not in the document]
Comment (nattikim) [reply backend/server/test/get_test.go:28]: @maximihajlov  maybe it should be done by those who's doing the tests?

File: frontend/src/styles/globals.css [hunk 7]
```diff
@@ -24,3 +24,23 @@ body {
...
+  .clickable {
+    @apply hover:opacity-50;
+  }
+
+  .button {
+    @apply text-white bg-blue-700 hover:bg-blue-800 focus:ring-4 focus:outline-none focus:ring-blue-300 font-medium rounded-lg text-sm w-full sm:w-auto px-5 py-2.5 text-center dark:bg-blue-600 dark:hover:bg-blue-700 dark:focus:ring-blue-800;
+  }
+
+  .inputbox-title {
+    @apply my-1;
```
[the remark this answers is not in the document]
Comment (nattikim) [reply frontend/src/styles/globals.css:90]: @maximihajlov this should be part of Design changes. In new branch

File: frontend/src/pages/create.tsx
Comment (maximihajlov) [inline frontend/src/pages/create.tsx:1]: No client-side validation for fields length and spaces (no trim)

**Review by maximihajlov — CHANGES_REQUESTED** [review]

**Review by maximihajlov — APPROVED** [review]

**Pull request merged**